pkg/account/persistence/loader/load.go

```diff
@@ -46,31 +46,33 @@ func LoadResources(fs afero.Fs, workingDir string, projects manifest.ProjectDefi
 		}
 		for _, pol := range res.Policies {
 			if _, exists := resources.Policies[pol.ID]; exists {
-				return nil, fmt.Errorf("policy with id %q already defined in another project", pol.ID)
+				return nil, fmt.Errorf("policy with id '%s' already defined in another project", pol.ID)
 			}
 			resources.Policies[pol.ID] = pol
 		}
 
 		for _, gr := range res.Groups {
 			if _, exists := resources.Groups[gr.ID]; exists {
-				return nil, fmt.Errorf("group with id %q already defined in another project", gr.ID)
+				return nil, fmt.Errorf("group with id '%s' already defined in another project", gr.ID)
 			}
 			resources.Groups[gr.ID] = gr
 		}
 
 		for _, us := range res.Users {
 			if _, exists := resources.Users[us.Email.Value()]; exists {
-				return nil, fmt.Errorf("user with email %q already defined in another project", us.Email)
+				return nil, fmt.Errorf("user with email '%s' already defined in another project", us.Email)
 			}
 			resources.Users[us.Email.Value()] = us
 		}
 
 		if featureflags.ServiceUsers.Enabled() {
 			for _, su := range res.ServiceUsers {
-				if _, exists := resources.ServiceUsers[su.Name]; exists {
-					return nil, fmt.Errorf("service user with name %q already defined in another project", su.Name)
+				for _, existingServiceUser := range resources.ServiceUsers {
+					if err := verifySerivceUsersAreNotAmbiguous(su, existingServiceUser); err != nil {
```

Comment:
Just to get it right, we are using `verifySerivceUsersAreNotAmbiguous` two times, one to check if it's unique for one project and one for checking if it's unique for all projects.
This means we actually don't need to check it within `Load` if we change it to be private (no usages besides tests).

Because the check is done across projects it would make sense to also mention the project in the error message

Reply:
I would suggest an improvement in error handling here in general. It may make sense to introduce an error type that gets the project. Also splitting the big for loop into smaller methods, so we have less complexity.

Reply:
Totally agree - it is already in the (big) list of follow ups 👍

Reply:
I have improved the error messages through a small refactoring, but would like to clarify the role of projects in account management before doing too much more.